modules/internal/ChapelArray.chpl

```diff
@@ -1736,7 +1736,8 @@ module ChapelArray {
        instance of ``val`` in the array, or if ``val`` is not found, a
        tuple containing ``false`` and an unspecified value is returned.
      */
-    proc find(val: this.eltType): (bool, index(this.domain)) {
+     deprecated "find is deprecated, use a reduction like 'maxloc reduce zip(A == val, A.domain)' instead"
```

Comment:
Oh, was there not a test of this deprecation message?  I think there should be

Reply:
You're right, I've added this test to test/deprecated.

```chpl
var A = [1, 2, 1, 3, 1, 4, 1, 5];
writeln("Count of 1's in A = ", A.count(1));
writeln("A sorted is = ", A.sorted());
writeln("Finding 3 in A = ", A.find(3));
A.reverse();
writeln("Reversing A: ", A);
```

I'll need update the .good file as I merge in the related commits.

Although I think to make things easier I'll probably end up closing these PRs, merging them into one unified PR and merging that. I can keep things in separate commits for software archaeology purposes.

Reply:
That's probably the easiest way to handle it and I don't have a problem with it :)